### PR TITLE
Fix Parsing issue where there are spaces at the end of the quoted fields

### PR DESCRIPTION
**Pull request by smitshah91**
Added a fix where there are spaces at the end of the quoted fields. Also added tests. Issue #452

### CODE REVIEW

File: papaparse.js [hunk 1]
```diff
@@ -1390,15 +1390,15 @@
 			}
 
 			/**
-             * checks if there are extra spaces after closing quote and newLine without any text
+             * checks if there are extra spaces after closing quote and given index without any text
              * if Yes, returns the number of spaces
             */
-            function extraSpacesLengthAfterQuote(index) {
```
Comment (smitshah91) [inline papaparse.js]: Apologies for formatting issue. looks good in multiple IDEs I tried. I will appreciate your help in this one.

File: papaparse.js [hunk 2]
```diff
@@ -1390,15 +1390,15 @@
...
+            function extraSpaces(index) {
                 var spaceLength = 0;
 				if (index !== -1) {
-                    var textBetweenQuoteAndNextNewLine = input.substring(quoteSearch + 1, index);
-					if (textBetweenQuoteAndNextNewLine && textBetweenQuoteAndNextNewLine.trim() == '') {
```
Comment (smitshah91) [inline papaparse.js]: Same as above. apologies for formatting.

**Comment by pokoli** [comment]
After downloading the code I don't see any formating issues, so probably it's something related to how github is showing the code difffs. Forget about it. 

I merged it manually on ec7653dae70479104f8861fcc43317283add0960 I've squashed both commits and rewritten commit message to be more clear. 

Thanks for your contribution.

**Comment by smitshah91** [comment]
Thanks a lot :)

**Pull request closed without merging**